.gitignore

```diff
@@ -1,7 +1,28 @@
+# OS Specific
 .DS_Store
+.DS_Store?
+._*
+.Spotlight-V100
+.Trashes
+ehthumbs.db
+Thumbs.db
+
+# IDE Specific
+nbproject
+.~lock.*
+.buildpath
+.idea
+composer.lock
+*.sublime-workspace
+*.swp
+*.swo
+/.settings/
+/.project
```

Comment:
I'm personally not a fan of adding environment specific ignores to projects. Imho this should be done in the environment itself with a global gitignore. 
Also compose.lock will never appear, as we are not a PHP project ;)